sam_bot_description/src/description/sam_bot_description.sdf

```diff
@@ -0,0 +1,302 @@
+              <length>${wheel_width}</length>
+            </cylinder>
+          </geometry>
+          <material>
+            <ambient>0.3 0.3 0.3 1.0</ambient>
+            <diffuse>0.7 0.7 0.7 1.0</diffuse>
+          </material>
+        </visual>
+
+        <collision name="${prefix}_link_collision">
+          <pose relative_to="${prefix}_link">0 0 0 ${pi/2} 0 0</pose>
+          <geometry>
+            <cylinder>
+              <radius>${wheel_radius}</radius>
+              <length>${wheel_width}</length>
+            </cylinder>
+          </geometry>
+        </collision>
+
+        <xacro:cylinder_inertia m="0.5" r="${wheel_radius}" h="${wheel_width}"/>
+      </link>
+
+      <joint name="${prefix}_joint" type="revolute">
+        <parent>base_link</parent>
+        <child>${prefix}_link</child>
+        <pose relative_to="base_link">${x_reflect*wheel_xoff} ${y_reflect*(base_width/2+wheel_ygap)} ${-wheel_zoff} 0 0 0</pose>
+        <axis>
+          <xyz>0 1 0</xyz>
+          <limit>
+            <lower>-inf</lower>
+            <upper>inf</upper>
+          </limit>
+        </axis>
+      </joint>
+    </xacro:macro>
+
+    <xacro:wheel prefix="drivewhl_l" x_reflect="-1" y_reflect="1" />
+    <xacro:wheel prefix="drivewhl_r" x_reflect="-1" y_reflect="-1" />
+
+    <link name="front_caster">
+      <pose relative_to="caster_joint"/>
+
+      <visual name="front_caster_visual">
+        <geometry>
+          <sphere>
+            <radius>${(wheel_radius+wheel_zoff-(base_height/2))}</radius>
+          </sphere>
+        </geometry>
+        <material>
+          <ambient>0 1 1 1</ambient>
+          <diffuse>0 1 1 1</diffuse>
+        </material>
+      </visual>
+
+      <collision name="front_caster_collision">
+        <geometry>
+          <sphere>
+            <radius>${(wheel_radius+wheel_zoff-(base_height/2))}</radius>
+          </sphere>
+        </geometry>
+      </collision>
+
+      <xacro:sphere_inertia m="0.5" r="${(wheel_radius+wheel_zoff-(base_height/2))}"/>
+    </link>
+
+    <joint name="caster_joint" type="fixed">
+      <parent>base_link</parent>
+      <child>front_caster</child>
+      <pose relative_to="base_link">${caster_xoff} 0.0 ${-(base_height/2)} 0 0 0</pose>
+    </joint>
+
+    <joint name='imu_joint' type='fixed'>
+      <parent>base_link</parent>
+      <child>imu_link</child>
+      <pose relative_to="base_link">0.0 0.0 0.01 0 0 0</pose>
+    </joint>
+
+    <link name='imu_link'>
+      <must_be_base_link>true</must_be_base_link>
+      <visual name="imu_link_visual">
+        <geometry>
+          <box><size>
+            0.1 0.1 0.1
+          </size></box>
+        </geometry>
+      </visual>
+
+      <collision name="imu_link_collision">
+        <geometry>
+          <box><size>
+            0.1 0.1 0.1
+          </size></box>
+        </geometry>
+      </collision>
+
+      <xacro:box_inertia m="0.1" w="0.1" d="0.1" h="0.1"/>
+
+      <sensor name="imu_sensor" type="imu">
```

Comment:
I sthis publishing data? I think this needs to be something like 

```
<plugin filename="gz-sim-imu-system"
        name="gz::sim::systems::Imu">
</plugin>
```

Reply:
Fixed with [54b7e2e](https://github.com/ros-navigation/navigation2_tutorials/pull/102/commits/54b7e2ea85228ef297f8256ecf92971d20cecaef), thanks!